src/meta/src/hummock/manager/compaction_group_manager.rs

```diff
@@ -306,6 +306,7 @@ impl HummockManager {
                         .len() as u64
                         - 1,
                 );
+            new_version_delta.removed_table_ids.push(*table_id);
```

Comment:
Will it be too tricky if we reuse `GroupMetaChange` to do table id deletion so that we can avoid adding new fields to version delta? I took a brief look and it seems possible.

Reply:
How about reuse the `table_ids_remove` field in `GroupMetaChange` and clean `TableWatermark` when table delete

Reply:
> How about reuse the `table_ids_remove` field in `GroupMetaChange` and clean `TableWatermark` when table delete

Not that simple. We need to deduce the actual removed table ids by checking all group deltas in one version delta.

Reply:
> Will it be too tricky if we reuse GroupMetaChange to do table id deletion so that we can avoid adding new fields to version delta? I took a brief look and it seems possible.

The `GroupMetaChange` represents metadata changed in each compaction group, but table watermark does not belong to any compaction group so I think adding a new field is clearer. Besides, I think it's useful to globally track the removed table id in the hummock version. Therefore I chose a new field separated to group delta.

Reply:
No strong preference. Both work to me.

Reply:
Both work to me, thanks for your explanation